src/compile/mark/rule.ts

```diff
@@ -13,23 +13,29 @@ export namespace rule {
 
     // TODO: support explicit value
 
-    // vertical
-    if (model.has(X)) {
+    if (model.has(X) && model.has(Y) && model.has(Y2)) { // vertical range
```

Comment:
Right now, there are four cases.    Two of them vertical, two of them horizontal.  

Instead of breaking down like this,  can you:

1) Add rule for initialize `orient` for `rule` mark [here](https://github.com/vega/vega-lite/blob/master/src/compile/config.ts#L31). 

2) Have top-level `if` determine the `orient`. Then split case within each type of orientation.    
   - If horizontal, see if `x` and `x2` are specified.  Otherwise, use the default.  (If only `x` is specified, set `x2` to 0.) 
   - Similar for `y` except that `y2`'s default should be `{group: 'height'}`.